recipes/zstd/all/test_package/conanfile.py

```diff
@@ -1,17 +1,26 @@
 import os
-from conans import ConanFile, CMake, tools
+
+from conan import ConanFile
+from conan.tools.cmake import CMake, cmake_layout
+from conan.tools.build import cross_building
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package"
+    generators = "CMakeDeps", "CMakeToolchain"
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
 
     def build(self):
         cmake = CMake(self)
         cmake.configure()
         cmake.build()
 
+    def layout(self):
+        cmake_layout(self)
```

Comment:
I've tested this layout in a test package, and I have to say it's not very convenient to end up with all these generated conan files at the same level than conanfile.py in test_package. I would prefer to isolate these files in another folder.

Reply:
Was that with Conan 1.x or 2.0-beta?

In 1.x you still need the `-if` to find those files https://docs.conan.io/en/latest/migrating_to_2.0/commands.html#conan-install

Reply:
It's with conan 1.x indeed. Thanks for the link.